packages/create-gator-app/templates/nextjs/starter/src/utils/delegationUtils.ts

```diff
@@ -1,27 +1,31 @@
 import {
+  createDelegation,
   createExecution,
-  createRootDelegation,
+  Delegation,
   DelegationFramework,
-  DelegationStruct,
   SINGLE_DEFAULT_MODE,
-} from "@metamask-private/delegator-core-viem";
+} from "@metamask/delegation-toolkit";
 import { Address, Hex } from "viem";
 
 export function prepareRootDelegation(
   delegator: Address,
   delegate: Address
-): DelegationStruct {
+): Delegation {
   console.log(delegate, delegator);
-  return createRootDelegation(delegate, delegator, []);
+  return createDelegation({
+    to: delegate,
+    from: delegator,
+    caveats: [],
```

Comment:
We should definitely either _not_ create delegations with empty caveats, or we should be really explicit about it being a terrible idea.

Ideally we would create restricted caveats, but if we can't for whatever reason, we should:
```
caveats: createCaveatBuilder({
  environment,
  allowEmptyCaveats: true // something something, it's really important to not do this. Make sure you specify sufficient caveats
})
```